Description: Use cblas instead of calls to non existing codtc and zdotc C routines
Author: Sebastien Fabbro <bicatali@gentoo.org>

diff -Nur magma-1.2.0.orig/src/clatrd2.cpp magma-1.2.0/src/clatrd2.cpp
--- magma-1.2.0.orig/src/clatrd2.cpp	2012-06-06 17:41:50.000000000 +0100
+++ magma-1.2.0/src/clatrd2.cpp	2012-06-06 17:45:23.000000000 +0100
@@ -12,6 +12,7 @@
 
 */
 #include "common_magma.h"
+#include <cblas.h>
 
 #define PRECISION_c
 
@@ -270,7 +271,9 @@
           blasf77_cscal(&i, &tau[i - 1], W(0, iw), &ione);
 
 #if defined(PRECISION_z) || defined(PRECISION_c)
-          blasf77_cdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione);
+          cblas_cdotc_sub(i, W(0, iw), ione, A(0, i), ione, &value);
+
+//          blasf77_cdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione);
           alpha = tau[i - 1] * -.5f * value;
 #else
           alpha = tau[i - 1] * -.5f * blasf77_cdotc(&i, W(0, iw), &ione, A(0, i), &ione);
@@ -349,7 +352,10 @@
                             W(0, i), &ione, &c_one, W(i+1, i), &ione);
               blasf77_cscal(&i_n, &tau[i], W(i+1,i), &ione);
               #if defined(PRECISION_z) || defined(PRECISION_c)
-                    blasf77_cdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
+                        cblas_cdotc_sub(i_n, W(i +1, i), ione,
+                                        A(i +1, i), ione, &value);
+                    
+                  //blasf77_cdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
                   alpha = tau[i]* -.5f * value;
               #else
                   alpha = tau[i]* -.5f* blasf77_cdotc(&i_n, W(i+1,i), &ione, A(i+1, i), &ione);
diff -Nur magma-1.2.0.orig/src/zlatrd2.cpp magma-1.2.0/src/zlatrd2.cpp
--- magma-1.2.0.orig/src/zlatrd2.cpp	2012-06-06 17:41:50.000000000 +0100
+++ magma-1.2.0/src/zlatrd2.cpp	2012-06-06 17:46:33.000000000 +0100
@@ -12,6 +12,7 @@
 
 */
 #include "common_magma.h"
+#include <cblas.h> 
 
 #define PRECISION_z
 
@@ -270,7 +271,9 @@
           blasf77_zscal(&i, &tau[i - 1], W(0, iw), &ione);
 
 #if defined(PRECISION_z) || defined(PRECISION_c)
-          blasf77_zdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione);
+          cblas_zdotc_sub(i, W(0, iw), ione, A(0, i), ione, &value);
+
+//          blasf77_zdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione);
           alpha = tau[i - 1] * -.5f * value;
 #else
           alpha = tau[i - 1] * -.5f * blasf77_zdotc(&i, W(0, iw), &ione, A(0, i), &ione);
@@ -349,7 +352,10 @@
                             W(0, i), &ione, &c_one, W(i+1, i), &ione);
               blasf77_zscal(&i_n, &tau[i], W(i+1,i), &ione);
               #if defined(PRECISION_z) || defined(PRECISION_c)
-                    blasf77_zdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
+                        cblas_zdotc_sub(i_n, W(i +1, i), ione,
+                                        A(i +1, i), ione, &value);
+                    
+                  //blasf77_zdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
                   alpha = tau[i]* -.5f * value;
               #else
                   alpha = tau[i]* -.5f* blasf77_zdotc(&i_n, W(i+1,i), &ione, A(i+1, i), &ione);
